File: Compiler/language-containers/javascript/runner.js
const express = require('express');
const bodyParser = require('body-parser');
const fs = require('fs');
const { exec } = require('child_process');

const app = express();
app.use(bodyParser.json());

app.post('/run', (req, res) => {
  const code = req.body.code;
  fs.writeFileSync('script.js', code);

  exec('node script.js', (err, stdout, stderr) => {
    res.json({ output: stdout, error: stderr });
  });
});

app.listen(5004, () => console.log('JavaScript container on 5004'));

File: Compiler/language-containers/cpp/runner.py
from flask import Flask, request
import subprocess

app = Flask(__name__)

@app.route('/run', methods=['POST'])
def run():
    code = request.json.get('code')
    with open("main.cpp", "w") as f:
        f.write(code)
    compile = subprocess.run(['g++', 'main.cpp', '-o', 'main'], capture_output=True, text=True)
    if compile.returncode != 0:
        return {'output': '', 'error': compile.stderr}
    result = subprocess.run(['./main'], capture_output=True, text=True)
    return {'output': result.stdout, 'error': result.stderr}

app.run(host='0.0.0.0', port=5002)
